webClient/src/app/core/menu-bar/menu-bar.component.ts

```diff
@@ -27,6 +27,29 @@ import { MessageDuration } from "../../shared/message-duration";
 import { DeleteFileComponent } from '../../shared/dialog/delete-file/delete-file.component';
 import { Angular2InjectionTokens } from 'pluginlib/inject-resources';
 
+function initMenu(menuItems) {
+  menuItems.forEach(function(menuItem) {
```

Comment:
Was there a good reason for moving this outside the component, instead of just creating two component methods and calling it in the constructor like all the others? I'm guessing initialization precedence over other things isn't the issue since you're still calling it in the constructor. There doesn't look anything inherently wrong with this, I just haven't seen it used in the Editor code before

Reply:
Honestly wish we did more of this sort of thing: I was trying to keep functions private that really have no reason to be public. Typescript "private" keyword isnt actually private at runtime from my understanding.